airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/BaseConnector.java

```diff
@@ -42,7 +44,18 @@
   public ConnectorSpecification spec() throws Exception {
     // return a JsonSchema representation of the spec for the integration.
     final String resourceString = MoreResources.readResource("spec.json");
-    return Jsons.deserialize(resourceString, ConnectorSpecification.class);
+    return Jsons.object(addToSpec(Jsons.deserialize(resourceString)), ConnectorSpecification.class);
+  }
+
+  /**
+   * Extension point for child classes to add things to the spec json tree before it's deserialized
+   * into a ConnectorSpecification object.
+   *
+   * @param root
+   * @return root
+   */
+  public JsonNode addToSpec(JsonNode root) throws IOException {
```

Comment:
```suggestion
  public JsonNode transformSpec(JsonNode root) throws IOException {
```

Reply:
Would it makes sense to not expose this method in `BaseConnector` and instead just doing it on `AbstractJdbcDestination` and `AbstractJdbcSource`? I know it means that we will have to override `spec()` in those respective classes, so it probably a tiny bit more code change. On the plus side though, I think this transformation is going to be pretty specific to our JDBC sources and destinations, so exposing this transform just in those abstractions as opposed to all of our other connectors keeps the base a bit simpler.